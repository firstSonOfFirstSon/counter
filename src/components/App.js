import React from 'react';

const incIfOdd = (counter, increment) => {
   if (counter % 2 !== 0) {
      increment();
   }
}

const incAsync = (counter, increment) => {
   setTimeout(increment, 1000);
}

const App = ({counter = 0, increment = f => f, decrement = f => f}) => (
      <div className="app">
         Clicked: {counter} times
         <button onClick={increment}>+</button>
         <button onClick={decrement}>-</button>
         <button onClick={ () => incIfOdd(counter, increment) }>increment if odd</button>
         <button onClick={ () => incAsync(counter, increment) }>increment async</button>
      </div>
)
export default App;