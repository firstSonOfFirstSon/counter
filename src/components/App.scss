$light-grey: rgba(229, 229, 229, 1);
$light: rgba(255, 255, 255, 1);

@mixin shadow {
  box-shadow: 3px 3px 8px 1px darken($light-grey, 20%);
}

@mixin fade-bg($from, $to) {
  background: $from;
  background: linear-gradient(to bottom, $from 0%, $to 100%);
}

@mixin clear {
  margin: 0;
  padding: 0;
}

html, body, #root {
  height: 100%;
  min-height: 100%;
}

body {
  @include clear;
  @include fade-bg($light, $light-grey);
}

.app {
  padding: 20px;
}