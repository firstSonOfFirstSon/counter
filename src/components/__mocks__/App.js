import React from 'react';

const App = () => <div className="app-mock"></div>;

App.displayName = 'AppMock';

export default App;