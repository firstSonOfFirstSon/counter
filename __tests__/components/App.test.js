import React from 'react';
import App from "../../src/components/App";
import enzyme from 'enzyme';
import { JSDOM } from 'jsdom';
import toJSON from 'enzyme-to-json'
import { compose } from 'redux';
import Adapter from 'enzyme-adapter-react-16';


const { window } = new JSDOM(`...`);
const { document } = (new JSDOM(`...`)).window;

enzyme.configure({ adapter: new Adapter() });
global.window = window;
global.document = document;

describe('<App /> ui component', () => {

   describe('Rendering ui', () => {

      it('Renders Correctly', () => {
         compose(expect, toJSON, enzyme.shallow)(
            <App />
         ).toMatchSnapshot();
      });

      it('Renders div with class .app', () => {
         expect(
            enzyme.mount( <App /> ).
            find('div.app').
            first().length
         ).toBe(1)
      });

      it('Clicking on button does not cause errors', () => {
         enzyme.mount( <App /> ).
         find('button').
         first().
         simulate('click')
      });

   })

});