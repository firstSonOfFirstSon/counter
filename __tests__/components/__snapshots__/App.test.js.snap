// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> ui component Rendering ui Renders Correctly 1`] = `
<div
  className="app"
>
  Clicked: 
  0
   times
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    increment if odd
  </button>
  <button
    onClick={[Function]}
  >
    increment async
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
</div>
`;
